src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -653,6 +663,33 @@ public void finish() {
             return coverageDistribution;
         }
 
+        /** Emits a file with per base coverage if an output file has been set. */
+        private void emitPerBaseCoverageIfRequested() {
+            if (this.perBaseOutput == null) return;
```

Comment:
This is where Option would be nice (despite @ktibbett admitting early disdain of it!) because you could just foreach on the value. Alas.

I'm not a huge fan of the "call a function which maybe actually does something" type of function (now watch me have a bunch of them in my git history) vs. "call this function if you want it to do something", but I don't care strongly enough to hold this up.